hydromt/components/geoms.py

```diff
@@ -0,0 +1,171 @@
+
+    @hydromt_step
+    def read(self, filename: Optional[str] = None, **kwargs) -> None:
+        r"""Read model geometries files at <root>/<filename>.
+
+        key-word arguments are passed to :py:func:`geopandas.read_file`
+
+        Parameters
+        ----------
+        filename : str, optional
+            filename relative to model root. should contain a {name} placeholder
+            which will be used to determine the names/keys of the geometries.
+            if None, the path that was provided at init will be used.
+        **kwargs:
+            Additional keyword arguments that are passed to the
+            `geopandas.read_file` function.
+        """
+        self._root._assert_read_mode()
+        self._initialize(skip_read=True)
+        read_path = join(self._root.path, filename or self._filename)
+        fn_glob, _, regex = ConventionResolver()._expand_uri_placeholders(read_path)
+        fns = glob(fn_glob)
+        for fn in fns:
+            name = ".".join(regex.match(fn).groups())  # type: ignore
+            geom = cast(GeoDataFrame, gpd.read_file(fn, **kwargs))
+            self._logger.debug(f"Reading model file {name} at {fn}.")
+
+            self.set(geom=geom, name=name)
+
+    @hydromt_step
+    def write(
+        self,
+        filename: Optional[str] = None,
+        to_wgs84: bool = False,
+        **kwargs,
+    ) -> None:
+        r"""Write model geometries to a vector file (by default GeoJSON) at <root>/<filename>.
+
+        key-word arguments are passed to :py:meth:`geopandas.GeoDataFrame.to_file`
+
+        Parameters
+        ----------
+        filename : str, optional
+            filename relative to model root. should contain a {name} placeholder
+            which will be used to determine the names/keys of the geometries.
+            if None, the path that was provided at init will be used.
+        to_wgs84: bool, optional
+            If True, the geoms will be reprojected to WGS84(EPSG:4326) before they are written.
+        **kwargs:
+            Additional keyword arguments that are passed to the
+            `geopandas.to_file` function.
+        """
+        self._root._assert_write_mode()
+
+        if len(self.data) == 0:
+            self._logger.debug("No geoms data found, skip writing.")
+            return
+
+        for name, gdf in self.data.items():
+            if len(gdf) == 0:
+                self._logger.warning(f"{name} is empty. Skipping...")
+                continue
+
+            geom_filename = filename or self._filename
+
+            write_path = Path(
+                join(
+                    self._root.path,
+                    geom_filename.format(name=name),
+                )
+            )
+
+            self._logger.debug(f"Writing file {write_path}")
+
+            write_folder = dirname(write_path)
+            if not isdir(write_folder):
+                os.makedirs(write_folder, exist_ok=True)
+
+            if to_wgs84:
```

Comment:
This is only to respect GeoJSON file conventions so this is why we would only like to apply it for geojson files and not other extension as you would be loosing the original CRS which is quite dangerous...
So could add gain the old checks on geojson as well?

```suggestion
            if to_wgs84 and (
                kwargs.get("driver") == "GeoJSON"
                or str(fn).lower().endswith(".geojson")
            ):
```